ClockDisplay.java

```diff
@@ -49,7 +49,15 @@ public void timeTick()
     {
         minutes.increment();
         if(minutes.getValue() == 0) {  // it just rolled over!
-            hours.increment();
+            if(hours.getValue() == 12)
+            {
+                hours.increment();
+                hours.increment();
```

Comment:
This is a kluge required by changing the way the numberDisplay is expected to work. You should leave the number display dealing with 0..11 and modify the method that handles what we show the user.